pkg/collector/osmlogs_collector.go

```diff
@@ -171,6 +161,7 @@ func collectEnvoyData(collector *OsmLogsCollector, rootPath, meshName, namespace
 
 // ** Collects logs of every OSM controller in a given mesh **
 func collectControllerLogs(collector *OsmLogsCollector, rootPath, meshName string) error {
+	rootPath = filepath.Join(rootPath, "controller_logs")
 	controllerInfos, err := utils.GetResourceList([]string{"get", "pods", "--all-namespaces", "-l", "app=osm-controller", "-o", "custom-columns=NAME:{..metadata.name},NAMESPACE:{..metadata.namespace}"}, "\n")
```

Comment:
`"-l", "app=osm-controller,meshName=" + meshName` selector to filter and get only the controllers belonging to this mesh.

Reply:
Oops this second suggestion doesn't work.

Reply:
The extended label only works for deployments (my suggestion on top).

Reply:
"-l", "app=osm-controller,app.kubernetes.io/instance=" + meshName

Reply:
I have replaced this function entirely with `collectPodLogs` so it now collects logs from pods in a specific namespace. This is called on controller namespaces...theoretically there should be a 1:1 mapping of controller namespace to mesh so do you think this is resolved?

Reply:
Yeah it makes sense. `collectPodLogs` uses `kubectl get pods -n controllerNS`, so it's fine.